docs/config.rst

```diff
@@ -7,18 +7,19 @@ Configuration
 Configuration file
 ==================
 
-When a pipeline script is launched Nextflow looks for a file named ``nextflow.config`` in the current directory and
-in the script base directory (if it is not the same as the current directory). Finally it checks for the file
-``$HOME/.nextflow/config``.
-
-When more than one on the above files exist they are merged, so that the settings in the first override the same ones
-that may appear in the second one, and so on.
-
-The default config file search mechanism can be extended proving an extra configuration file by using the command line
-option ``-c <config file>``.
-
-.. note:: It's worth noting that by doing this, the files ``nextflow.config`` and ``$HOME/.nextflow/config`` are not
-  ignored and they are merged as explained above.
+When a pipeline script is launched Nextflow looks for configurations in multiple locations. Since multiple files can
+contain conflicting settings the sources are ranked to decide which setting to come in play. The possible sources for
+configuration values are listed below with the highest preference at the top:
+
+1. Command line individual parameter (``--something value``)
+2. Command line specified config file (``-c my_config``)
+3. A file named ``nextflow.config`` in the current directory
+4. A file named ``nextflow.config`` in the script base directory (when it is not the same as the current directory)
+5. The file ``$HOME/.nextflow/config``
```

Comment:
Is this right @pditommaso? That the pipeline default config trumps the user's home directory config? I would have expected 4. and 5. to be the other way around here.

I think maybe it is right and we've discussed this before... My memory's not what it used to be.. 🤔

Reply:
Yes, it does. That's useful for user default settings, like mail SMTP server, etc

Reply:
Ok great! In that case I think that this PR is ready to go?

Reply:
And to confirm: this reads that any settings in the pipeline `nextflow.config` has priority over a config file in the home directory?

This still surprises me as email / other parameters are typically initialised as `false` in the pipeline `nextflow.config` file. Which would mean that they would effectively be impossible to have defaults for in the home directory config.

Reply:
> this reads that any settings in the pipeline nextflow.config has priority over a config file in the home directory

Yes, the logic is the local overrides the common defaults

> other parameters are typically initialised as false in the pipeline nextflow.config file

the pipeline config, can still be overridden by the local config, tho I get what you are suggesting, that the pipeline should be the root, then "home" config, etc